rewrite-java/src/main/java/org/openrewrite/java/tree/JavaCoordinatesMap.java

```diff
@@ -0,0 +1,21 @@
+package org.openrewrite.java.tree;
+
+import org.openrewrite.internal.lang.Nullable;
+
+public abstract class JavaCoordinatesMap<J2 extends J> {
```

Comment:
I don't think this parameterization gives us anything, because where we use coordinates in `JavaTemplate`, we'll be implementing `visitEach` where the input type is just `J` anyway. we'll wind up casting `Coordinates#tree` based on which `coordinate` it is at.

Reply:
done.